### PR TITLE
e2etests/jenkins client plugin/verifyservice

### PR DESCRIPTION
**Pull request by waveywaves**
### Description 
This PR contains tests for https://github.com/openshift/jenkins-client-plugin/pull/261
Will remove WIP tag after making final canges to the above given PR

### CODE REVIEW

**Comment by waveywaves** [comment]
cc @gabemontero

**Comment by gabemontero** [comment]
@waveywaves you've got to run `hack/update-generated-bindata.sh` since you are manipulating testdata yaml

**Comment by waveywaves** [comment]
> @waveywaves you've got to run `hack/update-generated-bindata.sh` since you are manipulating testdata yaml

Done

**Comment by gabemontero** [comment]
@akram @waveywaves you guys have hit a few infrastructure flakes lately, but the https://prow.svc.ci.openshift.org/view/gcs/origin-ci-test/pr-logs/pull/23476/pull-ci-openshift-origin-master-e2e-aws-builds/2655 run I believe exposed a problem with @waveywaves 's new test.

I believe that run includes @waveywaves 's latest commit ... pretty sure, not 100% sure as they were on the same date as the run and the timestamps are hard to pinpoint.

There is a stack trace in the stdout of the test run that should prove helpful.

Have you all seen it?

**Comment by waveywaves** [comment]
@gabemontero Checking this rn. I will update the commits accordingly.

**Comment by akram** [comment]
@gabemontero can you PTAL ?

File: test/extended/builds/pipeline_origin_bld.go [hunk 1]
```diff
@@ -272,6 +273,38 @@ var _ = g.Describe("[Feature:Builds][Feature:Jenkins][Slow] openshift pipeline b
...
+				err := oc.Run("new-app").Args("-f", verifyServiceClientPluginPipelinePath).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				err = exutil.WaitForDeploymentConfig(oc.KubeClient(), oc.AppsClient().AppsV1(), oc.Namespace(), "nodejs-example", 1, false, oc)
+				if err != nil {
+					exutil.DumpApplicationPodLogs("nodejs-example", oc)
+				}
+
+				brservices, err := exutil.StartBuildAndWait(oc, "jenkins-verifyservice-pipeline")
+				debugAnyJenkinsFailure(brservices, oc.Namespace()+"-jenkins-verifyservice-pipeline", oc, true)
```
Comment (gabemontero) [inline test/extended/builds/pipeline_origin_bld.go]: move this into the error block

admittedly you copied this from some other of the tests ... they should have moved it within the err block as well ... feel free to correct those if you like @waveywaves but I'm good either way

File: test/extended/testdata/verifyservice-pipeline-template.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
+apiVersion: v1
+kind: Template
+labels:
+  template: nodejs-example
+metadata:
+  annotations:
```
Comment (gabemontero) [inline test/extended/testdata/verifyservice-pipeline-template.yaml]: annotation is unnecessary  ... let's remove

File: test/extended/testdata/verifyservice-pipeline-template.yaml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    iconClass: icon-jenkins
+    tags: instant-app,jenkins
+  name: nodejs-example
+objects:
+- apiVersion: v1
+  kind: Service
+  metadata:
+    annotations:
```
Comment (gabemontero) [inline test/extended/testdata/verifyservice-pipeline-template.yaml]: annotation is unnecessary ... let's remove

File: test/extended/testdata/verifyservice-pipeline-template.yaml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    ports:
+    - name: web
+      port: 8080
+      targetPort: 8080
+    selector:
+      name: nodejs-example
+- apiVersion: v1
+  kind: Service
+  metadata:
+    annotations:
```
Comment (gabemontero) [inline test/extended/testdata/verifyservice-pipeline-template.yaml]: annotation is unnecessary ... let's remove

File: test/extended/testdata/verifyservice-pipeline-template.yaml [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    clusterIP: None
+    ports:
+    - port: 8080
+      targetPort: 8080
+    selector:
+      name: nodejs-example
+- apiVersion: v1
+  kind: DeploymentConfig
+  metadata:
+    annotations:
```
Comment (gabemontero) [inline test/extended/testdata/verifyservice-pipeline-template.yaml]: annotation is unnecessary ...let's remove

File: test/extended/testdata/verifyservice-pipeline-template.yaml [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+          - containerPort: 8080
+          readinessProbe:
+            httpGet:
+              path: /
+              port: 8080
+            initialDelaySeconds: 3
+            timeoutSeconds: 3
+    triggers:
+    - imageChangeParams:
+        automatic: true
```
Comment (gabemontero) [inline test/extended/testdata/verifyservice-pipeline-template.yaml]: image change trigger is unnecessary ... remove it, just leave the config change trigger

File: test/extended/testdata/verifyservice-pipeline-template.yaml [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+        from:
+          kind: ImageStreamTag
+          name: nodejs:latest
+          namespace: openshift
+      type: ImageChange
+    - type: ConfigChange
+- apiVersion: v1
+  kind: BuildConfig
+  metadata:
+    labels:
```
Comment (gabemontero) [inline test/extended/testdata/verifyservice-pipeline-template.yaml]: don't need the label

File: test/extended/builds/pipeline_origin_bld.go (hunk 1 above)
Comment (waveywaves) [reply test/extended/builds/pipeline_origin_bld.go]: I have moved all debugAnyJenkinsFailure calls to the respective error blocks as you suggested.

File: test/extended/testdata/verifyservice-pipeline-template.yaml (hunk 5 above)
Comment (waveywaves) [reply test/extended/testdata/verifyservice-pipeline-template.yaml]: Removed all annotations

File: test/extended/testdata/verifyservice-pipeline-template.yaml (hunk 6 above)
Comment (waveywaves) [reply test/extended/testdata/verifyservice-pipeline-template.yaml]: Removed the image trigger

File: test/extended/testdata/verifyservice-pipeline-template.yaml (hunk 7 above)
Comment (waveywaves) [reply test/extended/testdata/verifyservice-pipeline-template.yaml]: Removed the label

File: test/extended/testdata/verifyservice-pipeline-template.yaml [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+      type: Rolling
+    template:
+      metadata:
+        labels:
+          name: nodejs-example
+        name: nodejs-example
+      spec:
+        containers:
+        - env: []
+          image: 'nodejs:latest'
```
Comment (waveywaves) [inline test/extended/testdata/verifyservice-pipeline-template.yaml]: After removing the imagechage trigger, have added the image and the tag here.

Comment (waveywaves) [reply test/extended/testdata/verifyservice-pipeline-template.yaml]: The tests were failing with error regarding not being able to find the nodejs image. After adding the imagechange trigger seems to have made it better.

@gabemontero Do you have any suggestion regarding any other change to the image attribute ?

**Comment by akram** [comment]
/retest

again faced the test-cmd flake

File: test/extended/testdata/verifyservice-pipeline-template.yaml (hunk 8 above)
Comment (gabemontero) [reply test/extended/testdata/verifyservice-pipeline-template.yaml]: OK my recollections and what was needed/not needed were just faulty here @waveywaves  .... let's leave the ICT as you have it now / originally had it

apologies for the churn

the other reductions were useful

and perhaps some context ... there have been efforts in the past to minimize the size of the bindata as much as possible ... so when we introduce new yaml, we attempt to be diligent

**Comment by waveywaves** [comment]
@adambkaplan can you PTAL ? This needs the your approval. 

cc @gabemontero @akram

**Pull request merged**